stock_picking_reorder_lines/models/stock_move.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Author: Alexandre Fayolle
+#    Copyright 2013 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp import fields
+from openerp.models import Model
+
+
+class StockMove(Model):
+    _inherit = 'stock.move'
+    _order = 'date_expected desc, sequence, id'
+
+    sequence = fields.Integer(
+        default=10,
```

Comment:
That's not true. If you don't put default value, then it will be 0, and you'll get the same problem in a tree when editable=bottom (which is the case). Let me show the sequence to reproduce the error:

* Add some order lines. All get 0 as sequence.
* If you save, there's no problem, as the second order is by ID, so the order is respected.
* Then, you reorder any line, and automatically all lines from the move line gets a progressive sequence number (1, 2, 3...).
* If you add a new line, it will start with zero sequence, but in the UI, it will appear at the bottom.
* Then, when you save, this line will be reordered before the lines that has been reordered, provoking an inconsistence, because you are unawared of this change.